app/actors/ClusterShutdownActor.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016 Inscope Metrics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package actors;
+
+import akka.actor.Address;
+import akka.actor.Props;
+import akka.actor.UntypedActor;
+import akka.cluster.Cluster;
+import akka.cluster.ClusterEvent;
+import akka.cluster.Member;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Actor that subscribes to cluster events and triggers a CompletableFuture when it sees itself removed
+ * from the cluster.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ClusterShutdownActor extends UntypedActor {
```

Comment:
The active host updaters are cluster singletons.  Also might use cluster sharding (maybe?) for preliminary alert evaluation.  We had cluster shutdown before, this is just nicer.